File: from-2401-to-2500/2444-count-subarrays-with-fixed-bounds/solution.py
class Solution:
    def countSubarrays(self, nums: List[int], minK: int, maxK: int) -> int:
        bad = mi = ma = -1
        result = 0

        for i, n in enumerate(nums):
            if not minK <= n <= maxK:
                bad = i

            if minK == n:
                mi = i
            if maxK == n:
                ma = i

            start = min(mi, ma)
            if start > bad:
                result += (start - bad)

        return result
        

File: from-2401-to-2500/2444-count-subarrays-with-fixed-bounds/solution.go
func countSubarrays(nums []int, minK int, maxK int) int64 {
    bad, mi, ma := -1, -1, -1
    var result int64 = 0

    for i, n := range nums {
        if minK > n || n > maxK {
            bad = i
        }

        if minK == n {
            mi = i
        }
        if maxK == n {
            ma = i
        }
        start := min(mi, ma)
        if start > bad {
            result += int64(start - bad)
        }
    }
    return result
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}